e2e/teams.loggedin.spec.ts

```diff
@@ -0,0 +1,195 @@
+      ])
+
+      await expect(spinner).not.toBeVisible()
+      await expect(userPage.getByText('Invitation(s) Sent')).toBeVisible()
+    })
+
+    await test.step('user should view pending invitations', async () => {
+      await expect(userPage.getByText('Pending Invitations')).toBeVisible({
+        timeout: testInfo.timeout * 3, // add extra timeout, so backend can finish sending request
+      })
+      const row = userPage.getByRole('row', { name: adminUserName! })
+      await expect(row).toBeVisible()
+      await expect(row.getByText(INVITATION_MESSAGE)).toBeVisible()
+    })
+
+    await test.step('admin should accept team invitation', async () => {
+      await goToDashboard(adminPage)
+
+      await Promise.all([
+        expect(
+          adminPage.getByRole('heading', { name: 'Your Teams' }),
+        ).toBeVisible(),
+        adminPage.getByLabel('Teams').click(),
+      ])
+
+      // get row for this invitation
+      // ...in case multiple test suite users have invited admin user at the same time
+      const row = adminPage.getByRole('row', { name: TEAM_NAME })
+      await expect(row.getByText(INVITATION_MESSAGE)).toBeVisible()
+
+      await row.getByRole('button', { name: 'Join' }).click()
+      await expect(row).not.toBeVisible()
+    })
+
+    await test.step('admin should view team page', async () => {
+      const teamLink = adminPage.getByRole('link', { name: TEAM_NAME })
+
+      // handle case where occasionally two team links are shown
+      // ...before resolving to one team link
+      await expect
+        .poll(async () => {
+          return await teamLink.count()
+        })
+        .toBe(1)
+
+      await teamLink.click()
+
+      await expect(
+        adminPage.locator('h3').filter({
+          hasText: TEAM_NAME,
+        }),
+      ).toBeVisible()
+      await expect(adminPage.getByText('2 team members')).toBeVisible()
+
+      await expect(
+        adminPage.getByRole('heading', { name: 'Managers' }),
+      ).toBeVisible()
+      await expect(
+        adminPage.getByRole('link', { name: userName! }),
+      ).toBeVisible()
+
+      await expect(
+        adminPage.getByRole('heading', { name: 'Members' }),
+      ).toBeVisible()
+      await expect(
+        adminPage.getByRole('link', { name: adminUserName! }),
+      ).toBeVisible()
+    })
+  })
+
+  test.afterAll(async () => {
+    // delete team
+    await goToDashboard(userPage)
+    await userPage.getByLabel('Teams').click()
+    await userPage.getByRole('link', { name: TEAM_NAME }).click()
+    await userPage.getByRole('button', { name: 'Team Actions' }).click()
+    await userPage.getByRole('link', { name: 'Delete Team' }).click()
+    await userPage.getByRole('button', { name: 'Delete' }).click()
+    await expect(userPage.getByText('Team successfully deleted')).toBeVisible()
+    await expect(
+      userPage.getByRole('link', { name: TEAM_NAME }),
+    ).not.toBeVisible()
```

Comment:
Clean up the team in the `afterAll` hook, so that team is cleaned up even if the test fails part way through execution. Allows us to retry the test without cluttering the dev stack with teams.

If Teams are commonly used across tests, could consider creating a [Team fixture](https://playwright.dev/docs/test-fixtures#creating-a-fixture), which encapsulates creating and deleting a team.